<launch>
    <arg name="namespace" default="red"/>
    <arg name="carrot_out" default="carrot/trajectory"/>
    <arg name="potfield_in" default="carrot/trajectory"/>
    <arg name="potfield_out" default="potential_fields/trajectory"/>
    <arg name="mpc_in" default="potential_fields/trajectory"/>
    <arg name="odometry" default="msf_core/odometry"/>
    <arg name="model" default="bebop"/>

    <!-- MPC + MSF stack -->
    <include file="$(find uav_ros_control)/launch/bebop_mpc_msf.launch">
        <arg name="namespace" value="$(arg namespace)"/>
        <arg name="carrot_out" value="$(arg carrot_out)"/>
        <arg name="mpc_in" value="$(arg mpc_in)"/>
        <arg name="odometry" value="$(arg odometry)"/>
        <arg name="model" value="$(arg model)"/>
    </include>

    <!-- Potential fields -->
    <include file="$(find potential_fields)/launch/potential_fields.launch">
        <arg name="potfield_in" value="$(arg potfield_in)"/>
        <arg name="potfield_out" value="$(arg potfield_out)"/>
        <arg name="odometry" value="$(arg odometry)"/>
    </include>

    <!-- Velodyne to MPC state -->
    <node name="state_to_leddar" pkg="tf" type="static_transform_publisher" 
        args="0.1 0 0.15 0. 0.1323284641020683 3.141592653589793 state bebop/leddar 100" />
        
</launch>